docs/source/noaa_csp.rst

```diff
@@ -40,160 +51,252 @@ Login to the NOAA CSP
-above, it is strongly recommended the the user configure their
-respective experiments to use the ``/lustre`` file system for the
-``EXPDIR`` and ``ROTDIR`` contents. The ``/contrib`` file system is
-suitable for compiling and linking the workflow components required of
-the global-workflow.
+Assume you have a AWS cluster running, after login to the cluster through `ssh` from your laptop terminal,
+or access the cluster from your web terminal, one can start clone, compile, and run global-workflow.
+
+1. clone global-workflow( assume you have setup access to githup)::
+
+   cd /contrib/$USER   #you should have a username, and have a directory at /contrib where we save our permanent files.
+   git clone --recursive git@github.com:NOAA-EMC/global-workflow.git
+   #or the develop form at EPIC:
+   git clone --recursive git@github.com:NOAA-EPIC/global-workflow-cloud.git
+
+2. compile global-workflow::
+
+   cd /contrib/$USER/global-workflow
+   cd sorc
+   build_all.sh   # or similar command to compile for gefs, or others.
+   link_workflow.sh  # after build_all.sh finished successfully
+
+As users may define a very small cluster as controoler, one may use a script similar to this to compile in compute node.::
+
+   #!/bin/bash
+
+   #SBATCH --job-name=compile
+
+   #SBATCH --account=$USER
+
+   #SBATCH --qos=batch
+
+   #SBATCH --partition=compute
 
-The software stack supporting the ``develop`` branch of the
-global-workflow is provided for the user and is located beneath
-``/contrib/emc_static/spack-stack``. The modules required for the
-global-workflow execution may be loaded as follows.
+   #SBATCH -t 04:15:00
 
-.. code-block:: bash
+   #SBATCH --nodes=1
+
+   #SBATCH -o compile.%J.log
+
+   #SBATCH --exclusive
+
+   set -x
+
+   gwhome=/contrib/Wei.Huang/src/global-workflow-cloud
+   cd ${gwhome}/sorc
+   source ${gwhome}/workflow/gw_setup.sh
+
+   #build_all.sh
+
+   build_all.sh -w
+
+   link_workflow.sh
+
+Save the above lines in a file, say, com.slurm, and submit this job with command "sbatch com.slurm"
+
+3. run global-workflow C48 ATM test case (assume user has /lustre filesystem attached)::
+
+   cd /contrib/$USER/global-workflow
+   
+   HPC_ACCOUNT=${USER} pslot=c48atm RUNTESTS=/lustre/$USER/run \
+        ./workflow/create_experiment.py \
+        --yaml ci/cases/pr/C48_ATM.yaml
 
-   user@host:$ module unuse /opt/cray/craype/default/modulefiles
-   user@host:$ module unuse /opt/cray/modulefiles
-   user@host:$ module use /contrib/emc_static/spack-stack/miniconda/modulefiles/miniconda
-   user@host:$ module load py39_4.12.0
-   user@host:$ module load rocoto/1.3.3
+   cd /lustre/$USER/run/EXPDIR/c48atm
+   crontab c48atm
 
+EPIC has copied the common used data to AWS, and the current code has setup to use those data.
+If user wants to run own case, user needs to make changes to the IC path and others to make it work.
 The execution of the global-workflow should now follow the same steps
 as those for the RDHPCS on-premise hosts.
```

Comment:
This line should say "on-premises" with an "s", not "on-premise". Even when talking about only one location, it has an s.